dev-docs/RFCs/vNext/composite-layer-customization-rfc.md

```diff
@@ -40,62 +40,61 @@ const customizedGeoJsonLayer = new GeoJsonLayer({
   ...,
   transferAllProps: true,
   subLayerProps: {
-    ScatterplotLayer: {
-      layer: ExtendedScatterplotLayer,
+    points: {
+      type: ExtendedScatterplotLayer,
       extendedProp: 100
     },
-    PolygonLayer: {
+    'polygons-fill': {
       subLayerProps: {  // Recursive application of sublayerProps!
         SolidPolygonLayer: {
-          layer: HollowPolygonLayer
+          type: HollowPolygonLayer
         }
       }
     }
   }
 })
 ```
 
-In geojson:
-```js
-const defaultProps = {
-  ...,
-  subLayers: {
-    PointLayer: ScatterplotLayer,
-    LineLayer: PathLayer,
-    PolygonLayer: SolidPolygonLayer
-  }
-};
-
-const GeoJsonLayer extends CompositeLayer {
-  ...
-  renderLayers() {
-  	return [
-  	  new this.props.subLayerProps.PointLayer({
-  	  }),
-  	];
-  }
-}
-```
+#### Changes to the CompositeLayer API
 
+**getSubLayerProps**
 
-### Proposal: CompositeLayers take a `subLayerProps` prop
+The overriding props supplied by the user can be merged inside `compositeLayer.getSubLayerProps` so that each layer does not have to implement their own handling of `subLayerProps`.
 
+**getSubLayerClass**
+
+A new CompositeLayer class method that extracts the `type` field from `subLayerProps`, if any. This will replace the ad-hoc implementations in e.g. [HexagonLayer](https://github.com/uber/deck.gl/blob/6.3-release/modules/layers/src/hexagon-layer/hexagon-layer.js#L396) and [GeoJsonLayer](https://github.com/uber/deck.gl/blob/6.3-release/modules/layers/src/geojson-layer/geojson-layer.js#L71).
 
-In application:
-```js
-  const customizedGeoJsonLayer = new GeoJsonLayer({
-  	...,
-  	subLayerProps: {
-  	  PointLayer: ...
-  	}
-  })
-```
 
+#### Open Questions
 
+**Should keys be layer class name or sublayer id?**
```

Comment:
In the first proposal, it also mentioned  `There are two types of naming possible: ...`. Is it talking about the same issue as here?

https://github.com/uber/deck.gl/blob/4284dfa4c3c8b79b6f82f2a58a922b0de0eb8cda/dev-docs/RFCs/vNext/composite-layer-customization-rfc.md#proposal-compositelayers-document-which-sub-layers-they-render

Reply:
Yes